backend/controllers/user.js

```diff
@@ -1,48 +1,111 @@
 const { UnauthorizedError } = require("../helper/customErrors");
 const { bcryptHash } = require("../helper/bcrypt");
+const crypto = require('crypto');
+const nodemailer = require('nodemailer');
 
 //* Current User
 const currentUser = async (req, res, next) => {
   try {
-    const { loggedUser } = req;
-    if (!loggedUser) throw new UnauthorizedError();
+    const { loggedUser } = req
+    if (!loggedUser) throw new UnauthorizedError()
 
-    loggedUser.dataValues.email = req.headers.email;
-    delete req.headers.email;
+    loggedUser.dataValues.email = req.headers.email
+    delete req.headers.email
 
-    res.json({ user: loggedUser });
+    res.json({ user: loggedUser })
   } catch (error) {
-    next(error);
+    next(error)
   }
-};
+}
 
 //* Update User
 const updateUser = async (req, res, next) => {
   try {
-    const { loggedUser } = req;
-    if (!loggedUser) throw new UnauthorizedError();
+    const { loggedUser } = req
+    if (!loggedUser) throw new UnauthorizedError()
 
     const {
       user: { password },
       user,
-    } = req.body;
+    } = req.body
 
     Object.entries(user).forEach((entry) => {
-      const [key, value] = entry;
+      const [key, value] = entry
 
-      if (value !== undefined && key !== "password") loggedUser[key] = value;
-    });
+      if (value !== undefined && key !== "password") loggedUser[key] = value
+    })
 
     if (password !== undefined || password !== "") {
-      loggedUser.password = await bcryptHash(password);
+      loggedUser.password = await bcryptHash(password)
     }
 
-    await loggedUser.save();
+    await loggedUser.save()
 
-    res.json({ user: loggedUser });
+    res.json({ user: loggedUser })
   } catch (error) {
-    next(error);
+    next(error)
   }
-};
+}
 
-module.exports = { currentUser, updateUser };
+//* Request Password Reset
+const requestPasswordReset = async (req, res, next) => {
+  try {
+    const { email } = req.body
+    const user = await User.findOne({ where: { email } })
+    if (!user) throw new NotFoundError("User")
+
+    const token = crypto.randomBytes(20).toString('hex')
+    user.resetPasswordToken = token
+    user.resetPasswordExpires = Date.now() + 3600000 // 1 hour
+    await user.save()
+
+    const transporter = nodemailer.createTransport({
+      service: 'Gmail',
+      auth: {
+        user: process.env.EMAIL_ADDRESS,
+        pass: process.env.EMAIL_PASSWORD,
```

Comment:
Sensitive data such as email credentials should not be hardcoded. Consider using environment variables for `process.env.EMAIL_ADDRESS` and `process.env.EMAIL_PASSWORD` to enhance security.